src/js/hooks/useDiscordData.js

```diff
@@ -0,0 +1,38 @@
+import { useContext, useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+import { GlobalContext } from '../contexts/GlobalContextWrapper';
+
+const useDiscordData = (resource, userType) => {
+  const history = useHistory();
+  const { DiscordStore } = useContext(GlobalContext);
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
+    // A flag to make sure state does not change when the component is not mounted
+    let isMounted = true;
+
+    DiscordStore.getData(resource, userType).then((res) => {
+      if (isMounted) {
+        setData(res);
```

Comment:
This is a pretty opinionated line of code. Hopefully it doesn't limit you too much for future components